### ISSUE TITLE
Run Google Colab Dreambooth training on Vast.ai or Runpod.io

### ISSUE THREAD

**Issue by bach777** [TheLastBen/fast-stable-diffusion#518]
Due to the limits of google colab free tier, it is not viable the training in resolutions higher than +768x768, or even in 512x512, I have understood that it requires many steps to obtain impeccable results, so I was wondering if there is a way to use your google colab template in a gpu rental page, or to do it in a local Colab session but connected to a rented gpu. Is it possible?
Thank you very much for your kind assistance!

**Comment by GuruVirus** [comment]
Yes. AItrepreneur has plenty of Runpod tutorials. https://www.youtube.com/c/Aitrepreneur/videos

**Comment by wktra** [comment]
Runpod costs an arm and a leg, and that's if a decent client is available. 
Google colab costs $10 a month.

**Comment by bach777** [comment]
the pricing for an RTX A5000 is $0.49/hr On-Demand

**Comment by bach777** [comment]
the price of a RTX A5000 is $0.49/hour on demand

**Comment by wktra** [comment]
Each training takes me about 4 hours for actual quality. That's about $2 right there. And then there are the disk charges ($4/mo Disk Charge). I train and tweak a few times a day. In less than a week, Ill be paying out the nose.

**Comment by bach777** [comment]
Thank you very much for the info, I will try to do it on Google Colab with the monthly tier. I hope they accept payments from my country

**Comment by wktra** [comment]
> Thank you very much for the info, I will try to do it on Google Colab with the monthly tier. I hope they accept payments from my country

I gotta be honest, I had trouble with my USA card when I tried to subscribe. I had to call my bank, tell them to not reject google colab and then try again.

**Comment by bach777** [comment]
Estoy teniendo un problema, estaba entrenando un modelo, (3000 steps) con la casilla de "resume training" sin haber marcado aun la de "Enable_text_encoder_training:" marcada, dos modelos con nombre A1-1000steps.ckpt y A1-2000steps.ckpt estan en mi disco de drive, Al intentar cargar la sesion dice 
"Previous model not found, training a new model..."
Y empieza a entrenar desde cero... Que deberia hacer?

**Comment by bach777** [comment]
I have trained a model (3000 steps) with "Continue training" , checked, and "Enable_text_encoder_training:" unchecked, two models named A1_step_1000.ckpt and A1_step_2000.ckpt are on my hard disk, It takes steps 4000-5000 steps to complete the training, right? . When I try to load the session, I get the message "Previous model not found, training a new model...". And it starts the training from the beginning.... What should I do?

**Comment by GuruVirus** [comment]
Just a warning, an acquaintance tested colab premium ($50) and Dreambooth would not run on the the provided hardware.

As far as restarting a session, is your session name the same?
![image](https://user-images.githubusercontent.com/4285187/202084205-24f7de1e-dcbe-4b97-8ef1-74b9acd13f69.png)

100 steps per image worked well with SD1.4, but 1.5 I haven't heard of people having good results worth reproducing.

**Comment by bach777** [comment]
![Screenshot (380)](https://user-images.githubusercontent.com/115612779/202085806-c454b8c8-b071-4504-a5e2-486baaa6efdc.png)
![Screenshot (381)](https://user-images.githubusercontent.com/115612779/202085815-8467f2ea-cf91-4fb3-83ac-dfe8d2b71105.png)
I referred as A1 as an example, this is the model that I try to train, apparently everything is correct but the training is restarting from 0

**Comment by TheLastBen** [comment]
The training is restarting because there is no final ckpt, only the intermediary checkpoint, if you want to resume anyway, rename one of the models to "sora.ckpt"

**Comment by emidio90** [comment]
I'm having problem running it on Paperspace, it's just stuck at cloning the repo

**Comment by 0xdevalias** [comment]
See also:

- https://github.com/TheLastBen/fast-stable-diffusion/issues/80
- https://github.com/TheLastBen/fast-stable-diffusion/pull/150
- https://github.com/TheLastBen/fast-stable-diffusion/issues/448
- https://github.com/TheLastBen/fast-stable-diffusion/issues/704

**Comment by SU1199** [comment]
idk if I'm too late but check this out https://github.com/SU1199/fastBooth
It's has all the performance modifies from the shivam's and theleastben notebooks with xformers.

**Comment by Shadhil24** [comment]
I tried running ShivamShivrao and ThelastBen on runpod and vast ai. Training in working fine but the model was not able to generate user given images after training. It is working fine in colab. What might be the reason. It would be a great help if anyone coult help me with this. Thank You

**Comment by TheLastBen** [comment]
@Shadhil24 I made a template for Runpod https://www.runpod.io/console/gpu-secure-cloud?template=runpod-stable-unified

**Comment by Shadhil24** [comment]
Can i run this on vast ai, sorry if this is a stupid question, i am new to
this

**Comment by TheLastBen** [comment]
The template is designed for runpod

**Comment by Shadhil24** [comment]
Is it necessary to given name for the images as same as the instance images

**Comment by Shadhil24** [comment]
Sorry for asking more questions, but how cam i give the same name to every
images, should i give count like shadhil_1.png, shadhil_2.png,
shadhil_3.png like that?

**Comment by TheLastBen** [comment]
yes, that's a correct format

**Comment by Shadhil24** [comment]
@TheLastBen Thank you, Its working fine now. I have changes my image names with the instance and class names

**Comment by TheLastBen** [comment]
don't use a class name though, only the token

**Comment by MohammadKatif** [comment]
ok
